app/grandchallenge/publications/templatetags/publication_extras.py

```diff
@@ -0,0 +1,43 @@
+from django import template
+
+from grandchallenge.algorithms.models import Algorithm
+from grandchallenge.archives.models import Archive
+from grandchallenge.challenges.models import Challenge, ExternalChallenge
+from grandchallenge.reader_studies.models import ReaderStudy
+
+register = template.Library()
+
+
+@register.filter
+def get_associated_objects(publication):

```

Comment:
Have you checked the number of requests that are being made for these views? My feeling is that this will produce a lot.

Reply:
Yes, this produces 5 queries per publication, so that explodes quickly. I'm not immediately sure how to reduce the queries, but I will look into it.